sympy/functions/combinatorial/factorials.py

```diff
@@ -876,10 +876,10 @@ def eval(cls, n, k):
         n, k = map(sympify, (n, k))
         d = n - k
         n_nonneg, n_isint = n.is_nonnegative, n.is_integer
-        if k.is_zero or ((n_nonneg or n_isint == False)
+        if k.is_zero or ((n_nonneg or n_isint is False)
```

Comment:
I am not sure of what `n.is_integer` or `n.is_nonnegative` will return in general. This is from [docs](http://docs.sympy.org/latest/modules/logic.html#sympy.logic.boolalg.BooleanTrue):

>  if there is any doubt over whether a function or expression will return S.true or True, just use == instead of is to do the comparison

Reply:
It also mentions:
> In other words, use S.true only on those contexts where the boolean is being used as a symbolic representation of truth.

I wonder if it is applicable to `is_integer` and `is_nonnegative`.

Reply:
I think they will be having three-valued logic:
> The assumptions system should use True and False. Aside from not satisfying the above rule of thumb, the assumptions system uses a three-valued logic (True, False, None), whereas S.true and S.false represent a two-valued logic. When in doubt, use True.